Roll back if `rollover::overwrite` fails when appending events to the DB

luckysori opened the pull request:
Previously we were allowing the following scenario:

- `RolloverCompleted` event is appended to the `events` table.
- `RolloverCompleted` event data fails to be appended to `rollover_completed_event_data` for whatever reason.

That scenario would leave us in a situation where the state between the two tables would be inconsistent.

To fix it, we ensure that the rollback logic doesn't just cover `rollover::overwrite::overwrite`, but rather the entirety of `append_event`.

Fixes https://github.com/itchysats/itchysats/issues/2662.

luckysori committed:
fix(sqlite-db): Roll back if rollover::overwrite fails

Previously we were allowing the following scenario:

- `RolloverCompleted` event is appended to the `events` table.
- `RolloverCompleted` event data fails to be appended to
  `rollover_completed_event_data` for whatever reason.

That scenario would leave us in a situation where the state between
the two tables would be inconsistent.

To fix it, we ensure that the rollback logic doesn't just cover
`rollover::overwrite::overwrite`, but rather the entirety of
`append_event`.

Fixes https://github.com/itchysats/itchysats/issues/2662.

da-kami reviewed (approved):
Nice! really like the refactor to avoid the need of the transaction. 
Happy to see this go in once the compile issues are resolved!

Restioson commented on sqlite-db/src/failed.rs:
```diff
@@ -256,11 +255,12 @@ async fn insert_failed_cfd(
-) -> Result<()> {
+async fn insert_event_log<'a, A>(conn: A, id: OrderId, event_log: EventLog) -> Result<()>
+where
+    A: Acquire<'a, Database = Sqlite>,
```
This refactor is nice! I don't really know exactly how this all works internally, but one thing I would note that in future, for some functions, `Transaction` might be preferable over `impl Acquire`, if the given operation should *always* be part of a bigger transaction, i.e if it could leave the db in an inconsistent state if a subsequent operation fails, if the function is designed only to be a helper function and always a constitutent of a larger operation.

I have no idea if this is ever true in this case, though :sweat_smile:

luckysori replied:
You raise an interesting point. On the one hand, by asking for a `Transaction` you inform the caller that they should be using one and therefore thinking about committing or rolling back. On the other hand, these functions don't need to interface with the `Transaction` API at all: we really don't want to call `commit` inside one of these functions (unless they are the very last thing that needs to be done)!

Given that we don't actually need to (or should) use the functionality provided by the trait inside these functions, I would lean towards passing in something that implements `Execute` (in this case we have to settle for `Acquire`, because of library limitation).

luckysori committed:
refactor(sqlite-db): Use SqliteConnection instead of Transaction

When we started to work with the `sqlx::Transaction` type to ensure
that certain database transactions were either fully committed or
rolled back, the type started leaking to lower level calls that should
not need to care about whether the caller is using an
`sqlx::Transaction` or a regular `sqlx::PoolConnection`.

By using the `sqlx::SqliteConnection` type in function type signatures
we are able to abstract over both `sqlx::Transaction`s and
`sqlx::PoolConnection`s. Because of this, we can greatly reduce the
boilerplate in `sqlite-db` tests.
